src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -1036,7 +1036,7 @@ private int processHeadCreate(SCMHeadEvent<?> event, TaskListener global, String
                         );
                         for (SCMHead h : event.heads(source).keySet()) {
                             String name = h.getName();
-                            Job job = p.getItem(name);
+                            Job job = p.getItemByBranchName(name);
```

Comment:
Not strictly necessary as getItem will pick up the old name, but removes the 1 in 33,554,432 chance if somebody for example has a branch called `feature/ux-1` and a branch called `feature-ux-1.0gorp7`